pg_variables.c

```diff
@@ -2116,6 +2116,14 @@ rollbackSavepoint(TransObject *object, TransObjectType type)
 {
 	TransState *state;
 
+	/* Nothing to do here if trans object was removed already. */
+	if (dlist_is_empty(&object->states))
+	{
+		/* Probably redundant. */
```

Comment:
Why is it redundant?

Reply:
As far as I can see, the only way we can get here is if variable is already removed. So all the tests are ok just with return here.